src/Server/src/Core/Data/IdentityServer/IdentityServerGroupService.cs

```diff
@@ -0,0 +1,79 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using IdOps.Model;
+using IdOps.Security;
+using IdOps.Server.Storage;
+
+namespace IdOps
+{
+    public class IdentityServerGroupService : UserTenantService, IIdentityServerGroupService
+    {
+        private readonly IIdentityServerGroupStore _store;
+    
+        public IdentityServerGroupService(
+            IUserContextAccessor userContextAccessor,
+            IIdentityServerGroupStore store)
+            :base(userContextAccessor)
+        {
+            _store = store;
+        }
+    
+        public async Task<IEnumerable<IdentityServerGroup>> GetGroupsByUserTenants(
+            CancellationToken cancellationToken)
+        {
+            IReadOnlyList<string> userTenants =await GetUserTenantsAsync(cancellationToken);
+            IEnumerable<IdentityServerGroup>? allGroups = await GetAllGroupsAsync(cancellationToken);
+    
+            var byTenant = new List<IdentityServerGroup>();
+    
+            foreach (IdentityServerGroup? group in allGroups)
+            {
+                if (group.Tenants.Any(userTenants.Contains))
+                {
+                    byTenant.Add(group);
+                }
+            }
+    
+            return byTenant;
+        }
+    
+        public Task<IReadOnlyList<IdentityServerGroup>> GetAllGroupsAsync(
+            CancellationToken cancellationToken)
+        {
+            return _store.GetAllAsync(cancellationToken);
+        }
+    
+        public async Task<IdentityServerGroup> SaveAsync(
+            SaveIdentityServerGroupRequest request,
+            CancellationToken cancellationToken)
+        {
+            var identityServerGroup = new IdentityServerGroup
+            {
+                Id = request.Id.GetValueOrDefault(Guid.NewGuid()),
```

Comment:
I dont know what `GetValueOrDefault` is, but ig uess something like request.Id === default ? value : request.Id
Theoretically a delegate would make sense here, as this way we initialize a Guid on every operation.

Reply:
Yep, it's exactly this:
```csharp
public readonly T GetValueOrDefault(T defaultValue) => !this.hasValue ? defaultValue : this.value;
```
we want to create a new Guid every time because it's a new group.